[Python] Invalid list access error with star in parameters

For the following program:
```python
class Foo:
    def __init__(self) -> None:
        pass

    def foo(self, *, l: list[str]):
        for i in l:
            pass
```
ESBMC errors out:
```bash
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:5: error: Function is missing a return type annotation  [no-untyped-def]
Found 1 error in 1 file (checked 1 source file)

Converting
ERROR: Invalid list access: could not resolve position or element type
```
If I remove the `*` from the parameters, verification succeeds.

ESBMC version: b912a8ef0

--- a/regression/python/github_3243/main.py
+++ b/regression/python/github_3243/main.py
@@ -1,0 +1,7 @@
+class Foo:
+    def __init__(self) -> None:
+        pass
+
+    def foo(self, *, l: list[str]):
+        for i in l:
+            pass

--- a/regression/python/github_3243/test.desc
+++ b/regression/python/github_3243/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+
+^VERIFICATION SUCCESSFUL$

--- a/regression/python/github_3243_2/main.py
+++ b/regression/python/github_3243_2/main.py
@@ -1,0 +1,20 @@
+class Foo:
+    def __init__(self) -> None:
+        pass
+    
+    def foo(self, *, l: list[str]):
+        assert len(l) > 0
+        
+        count = 0
+        for i in l:
+            assert isinstance(i, str)
+            assert i is not None
+            count += 1
+        
+        # Assert we iterated through all elements
+        assert count == len(l)
+
+f = Foo()
+test_list = ["hello", "world", "test"]
+f.foo(l=test_list)
+f.foo(l=["single"])

--- a/regression/python/github_3243_2/test.desc
+++ b/regression/python/github_3243_2/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+--unwind 9
+^VERIFICATION SUCCESSFUL$

--- a/regression/python/github_3243_2_fail/main.py
+++ b/regression/python/github_3243_2_fail/main.py
@@ -1,0 +1,20 @@
+class Foo:
+    def __init__(self) -> None:
+        pass
+    
+    def foo(self, *, l: list[str]):
+        assert len(l) > 0
+        
+        count = 0
+        for i in l:
+            assert isinstance(i, str)
+            assert i is None
+            count += 1
+        
+        # Assert we iterated through all elements
+        assert count == len(l)
+
+f = Foo()
+test_list = ["hello", "world", "test"]
+f.foo(l=test_list)
+f.foo(l=["single"])

--- a/regression/python/github_3243_2_fail/test.desc
+++ b/regression/python/github_3243_2_fail/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+--unwind 9
+^VERIFICATION FAILED$

--- a/src/python-frontend/json_utils.h
+++ b/src/python-frontend/json_utils.h
@@ -298,6 +298,16 @@ find_var_in_function(const std::string &var_name, const JsonType &func_node)
     }
   }
 
+  // Search in keyword-only arguments (after *)
+  if (func_node.contains("args") && func_node["args"].contains("kwonlyargs"))
+  {
+    for (const auto &arg : func_node["args"]["kwonlyargs"])
+    {
+      if (arg.contains("arg") && arg["arg"] == var_name)
+        return arg;
+    }
+  }
+
   // Then search in function body
   return get_var_node(var_name, func_node);
 }

